src/test/java/org/assertj/core/internal/spliterator/Spliterators_assertHasCharacteristics_Test.java

```diff
@@ -77,12 +78,18 @@ public void should_fail_if_spliterator_does_not_have_characteristic() {
     // GIVEN
     Spliterator<?> actual = createSpliterator(SORTED);
     // WHEN
-    expectAssertionError(() -> spliterators.assertHasCharacteristics(INFO, actual, DISTINCT));
+    AssertionError error = expectAssertionError(() -> spliterators.assertHasCharacteristics(INFO, actual, DISTINCT));
     // THEN
-    verify(failures).failure(INFO, shouldContain(singleton("SORTED"), array("DISTINCT"), singleton("DISTINCT")));
+    assertThat(error).hasMessage(format("%nExpecting spliterator characteristics:%n"
+                                 + " <[\"SORTED\"]>%n"
+                                 + "to contain:%n"
+                                 + " <[\"DISTINCT\"]>%n"
+                                 + "but could not find the following characteristics:%n"
+                                 + " <[\"DISTINCT\"]>%n"));
```

Comment:
this test (and the other similar ones) should not have changed, here we only want to verify that `shouldContain` is called with the proper arguments.
checking the actual message should be done in `ShouldContain_create_Test`

Reply:
After implementing ``getGroupTypeDescription`` method,
```java
Spliterator<?> actual = createSpliterator(SORTED);
spliterators.assertHasCharacteristics(INFO, actual, DISTINCT);
```
fails with this error
```
Expecting spliterator characteristics:
 <["SORTED"]>
to contain:
 <["DISTINCT"]>
but could not find the following characteristics:
 <["DISTINCT"]>
```
But
```java
shouldContain(singleton("SORTED"), array("DISTINCT"), singleton("DISTINCT"));
```
fails with this error
```
Expecting SingletonSet:
 <["SORTED"]>
to contain:
 <["DISTINCT"]>
but could not find the following elements:
 <["DISTINCT"]>
```
spliterator characteristics are converted to a Ser\<String\> in ``spliterators.assertHasCharacteristics`` method, so I use stack trace to check whether ``shouldContain`` method is called through Spliterator class. Thus, a direct calling of ``shouldContain`` method will display a different type name. And that's why I need to change this test